src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -33,31 +34,36 @@ public BookController(BookService bookService) {
     }
 
     @GetMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
you may simplify 
```suggestion
    @PreAuthorize("hasRole('USER')")
```

`hasRole` automatically adds `ROLE_` prefix

apply to all places